sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ClientRetryPolicy.java

```diff
@@ -367,6 +398,32 @@ private Mono<ShouldRetryResult> shouldRetryOnBackendServiceUnavailableAsync(
         return Mono.just(ShouldRetryResult.retryAfter(Duration.ZERO));
     }
 
+    private Mono<ShouldRetryResult> shouldRetryOnRequestTimeout(
+        boolean isReadRequest,
+        boolean nonIdempotentWriteRetriesEnabled) {
+
+        if (this.globalPartitionEndpointManagerForCircuitBreaker.isPartitionLevelCircuitBreakingApplicable(this.request)) {
+            if (!isReadRequest && !nonIdempotentWriteRetriesEnabled) {
```

Comment:
just curious, why only track the failure here for write & nonRetriable(even though for now only write operations can bubble 408), but we should track the exception for any operation type? (read & write?)

Reply:
Had a sync in the JVM meeting - conclusion was to scope it to transit timeout based `RequestTimeoutException` since there are cases where the client can generate one to (in the consistency-layer classes for example).